### PR TITLE
Addresses hashsolo pre_existing_cluster bugs

### PR DESCRIPTION
**Pull request by njbernstein**
Also added a check to make sure the number of noise barcodes is less than the number of sample barcodes.

Fixed the example in the doc string to include the cell_hashing_columns argument

### CODE REVIEW

**Comment by njbernstein** [comment]
https://github.com/theislab/scanpy/issues/1482

**Comment by adamgayoso** [comment]
@njbernstein I had also noticed hashsolo wasn't added to the documentation, which you can do by adding `pp.hashsolo` [here](https://github.com/theislab/scanpy/blob/master/scanpy/external/__init__.py). Though I'm unsure which heading it belongs under.

**Comment by njbernstein** [comment]
@adamgayoso  I don't think it fits under the other preprocessing tool headings of Data integration or Imputation.

Maybe add a new one called Call hashing or Sample demultiplexing. @fidelram thoughts? Not sure who else to tag

**Comment by adamgayoso** [comment]
@ivirshup

**Comment by ivirshup** [comment]
I think a new section would good here, maybe "Doublet detection"?

**Comment by njbernstein** [comment]
@ivirshup I think doublet detection is a little too narrow in that we can also associate cells to the original sample they came from. Although many folks use cell hashing solely for the purpose of doublet detection. I'd prefer "Sample demultiplexing".

**Comment by ivirshup** [comment]
I'd be happy with whatever, but it needs to go somewhere!

How about "Demultiplexing/ Doublet detection", so it can go with scrublet (#1476) in the next release, and potentially get split out later if more demultiplexing methods are added?

**Comment by njbernstein** [comment]
Any advice for getting the docs to build? The link above doesn't give any details.

File: scanpy/external/__init__.py [hunk 1]
```diff
@@ -34,6 +34,13 @@
    pp.harmony_integrate
    pp.mnn_correct
 
+Sample demultiplexing / Doublet detection
+~~~~~~~~~~~~~~~~
+.. autosummary::
```
Comment (adamgayoso) [inline scanpy/external/__init__.py:40]: add line between line 38 and 39

File: scanpy/external/__init__.py [hunk 2]
```diff
@@ -34,6 +34,13 @@
    pp.harmony_integrate
    pp.mnn_correct
 
+Sample demultiplexing / Doublet detection
+~~~~~~~~~~~~~~~~
```
Comment (adamgayoso) [inline scanpy/external/__init__.py]: the tildes should extend to the end of doublet detection

**Review by adamgayoso — COMMENTED** [review]
My comments will hopefully fix the docs

**Comment by njbernstein** [comment]
mm still not going. @adamgayoso sorry to bug ya again. or @ivirshup @fidelram is there a way to see why the docs are failing?

**Comment by ivirshup** [comment]
Could you rebase on master? I can't see any info on the docs building since there are conflicts.

Can you build the docs locally?

**Comment by njbernstein** [comment]
The docs build locally just fine. There are some warnings but it looks like they are from across the repo. Not sure why they are not building up here. The issue before was that I needed to remove typing from my docstrings. @adamgayoso @ivirshup

File: docs/api/scanpy.plotting.rst [hunk 3]
```diff
@@ -13,4 +13,11 @@ scanpy.plotting
...
-   
\ No newline at end of file
+   
+
+   
+   
+   
+
+
+
```
Comment (ivirshup) [inline docs/api/scanpy.plotting.rst:23]:
```suggestion
```

File: scanpy/external/pp/_hashsolo.py [hunk 4]
```diff
@@ -255,30 +252,30 @@ def hashsolo(
...
         for example leiden or cell types, not batches though
-    number_of_noise_barcodes : int,
+    number_of_noise_barcodes
         Use this if you wish change the number of barcodes used to create the
         noise distribution. The default is number of cell hashes - 2.
-    inplace : bool
+    inplace
         To do operation in place
     Returns
     -------
```
Comment (ivirshup) [inline scanpy/external/pp/_hashsolo.py]:
```suggestion
        To do operation in place

    Returns
    -------
```

File: scanpy/external/pp/_hashsolo.py [hunk 5]
```diff
@@ -255,30 +252,30 @@ def hashsolo(
         More information and bug reports `here <https://github.com/calico/solo>`__.
     Parameters
```
Comment (ivirshup) [inline scanpy/external/pp/_hashsolo.py]:
```suggestion
        More information and bug reports `here <https://github.com/calico/solo>`__.

    Parameters
```

**Review by ivirshup — CHANGES_REQUESTED** [review]
Hey @njbernstein, it looks like headings want an empty line ahead of them

**Comment by njbernstein** [comment]
finally! Thank you @ivirshup

**Review by ivirshup — APPROVED** [review]
Great, thanks for fixing that!

**Pull request merged**